react/week2/exersice-react/src/components-todo/TodoList.js

```diff
@@ -0,0 +1,31 @@
+import React, { useState } from 'react'
+import TodoItems from './TodoItems'
+import TodoListItem from './TodoListItem'
+
+function TodoList() {
+    const originalTodos = TodoItems();
+    let [updatedTodos, setTodos] = useState(originalTodos);
+
+    let newTodoDescription = '';
+
+    function addNewTodo() {
+        const id = updatedTodos.length + 1;
+        const date = new Date().toLocaleDateString("en-US", { weekday: 'long', year: 'numeric', month: 'long', day: 'numeric' });
+        const newTodo = { id: id, description: newTodoDescription, date: date }
+        updatedTodos = updatedTodos.concat(newTodo);
+        setTodos(updatedTodos)
+    }
```

Comment:
You can use useState for incoming inputs from user like  `const [userInput, setUserInput] = useState("");` instead of this newTodoDescription variable.
And also instead of this concat approach, I would recommend using the spread operator. Here is my approach:
```suggestion
  const addNewTodo = () => {
    const date = new Date().toLocaleDateString("en-US", {
      weekday: "long",
      year: "numeric",
      month: "long",
      day: "numeric",
    });

    setTodos([
      ...todos,
      {
        id: todos.length + 1,
        description: userInput,
        date: date,
      },
    ]);
  };
```
If you have no idea about this spread operator, I would recommend this [page](https://oprearocks.medium.com/what-do-the-three-dots-mean-in-javascript-bc5749439c9a) to read